.github/workflows/testing-mac.yml

```diff
@@ -26,15 +26,11 @@ jobs:
     - name: update package information
       run: brew update
     - name: install tools and libraries
-      # gdb and docutils need python@3.9.
-      # python@3.9 is not installed expectedly via gdb or docutils.
-      # Therefore we install libxml2 before installing before
-      # installing gdb and docutils. python@3.9 is installed well
-      # via libxml2.
       run: |
-        brew install autoconf automake pkg-config jansson libyaml bash
-        brew install libxml2 || brew link --overwrite libxml2
-        brew install gdb docutils
+        # libxml2 requires python@3.9.
+        # However, the request is not satisfied with just installing libxml2.
+        brew install libxml2 || brew link --overwrite python@3.9
+        brew install autoconf automake pkg-config jansson libyaml bash gdb docutils
```

Comment:
GitHub Actions VM `macOS10.15` `macOS11.0` have pre-installed `Python@3.9` via `HomeBrew`, the error happens when `Python@3.9` can be upgrade, so we should upgrade python@3.9 first.
```
brew upgrade python@3.9 || brew link --overwrite python@3.9

brew install libxml2 jansson libyaml autoconf automake pkg-config bash gdb docutils
```

Reply:
Thank you. You know so many things! I will make a pull request reflecting this suggestion.